data/transactions/logic/opcodes.go

```diff
@@ -184,6 +189,36 @@ func opBranch() OpDetails {
 	return d
 }
 
+func multiOp(dispatch byte, name string, rootVersion uint64, children ...OpSpec) OpSpec {
+	ret := OpSpec{Opcode: dispatch, Name: name, OpDetails: opDefault()}
+	ret.childOps = children
+	ret.Version = rootVersion
+	return ret
+}
+
+func isMultiOp(spec *OpSpec) bool {
+	return spec.childOps != nil
+}
+
+func getLeafSpecs(spec *OpSpec) []OpSpec {
+	if spec.childOps == nil {
+		return []OpSpec{*spec}
+	}
+	specs := make([]OpSpec, 0)
+	for _, child := range spec.childOps {
+		specs = append(specs, getLeafSpecs(&child)...)
+	}
+	return specs
+}
+
+// GetFullCode returns the full opcode of an OpSpec as a byte slice
+func GetFullCode(spec OpSpec) []byte {
```

Comment:
You can `Write()` this instead of the `if` in the assembler to decide to write the OpCode or MultiCode.